client/state/data-layer/wpcom-http/README.md

```diff
@@ -0,0 +1,140 @@
+# WordPress.com API HTTP Layer
+
+## Summary
+
+WordPress.com API requests require an initial translation from understanding what data pieces of Calypso might be wanting and the endpoints on WordPress.com which can provide that data.
+Once that translation has been made the requests have to be sent out over an HTTP connection.
+This middleware layer provides a foundation under the API data layer which can be used to pass around pure and stateless descriptions of those desired HTTP requests.
+By removing the burden of directly issuing and responding to HTTP requests in the API layer and moving it here, we can more easily accommodate global and centralized policies on those requests, such as batching, dropping, and retrying on failure.
+
+In contrast to a fully-generic HTTP layer this one understands that requests are being made directly to the WordPress.com API and allowances are made in the HTTP Request descriptions to acknowledge this and make working with it a smoother process; for example, authentication is automatically handled.
+
+## Why?
+
+The end-goal for this two-layer middleware approach is to make offline-use first-class in Calypso and to enable things like mobile data management and battery optimization.
+
+Let's look at how this system works from a high-level perspective.
+
+## Overview
+
+<!-- the following diagram was generated in draw.io - it can be edited by pasting in the contents of the SVG itself -->
+![](https://cldup.com/unQOzvDkjtq/vSrA37.svg)
+
+Wow, how confusing!
+Well, this isn't promising to make everything easier, but it _is_ promising to make things worth while.
+
+### Step A: Component needs to update data
+
+A React component wants to indicate that a given post ought to be marked as liked.
+It's API includes two functions for accomplishing this: `likePost( siteId, postId )` and `unlikePost( siteId, postId )`.
+It knows nothing about how those two functions work, only that they are the names it uses to accomplish its goal.
```

Comment:
Throughout the document, it might be good to highlight points like this which lend confidence in the process. Here, specifically that making the component unaware of the effects of these actions makes it more portable and future-proof by allowing the implementations of these props to be swapped out in the future without impacting the component itself.